SIG Observability Charter (#1)
cncf/sig-observability: observability-charter.md

[ddr]: https://github.com/cncf/toc/blob/master/process/due-diligence-guidelines.md
[sigwg]: https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#responsibilities--empowerment-of-sigs

## Governance

- This SIG follows the [standard operating model][som] provided by the TOC
  unless otherwise stated here.

[som]: https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#operating-model

## Operations

- The formation of the SIG follows the [documented process][sigform].

_Note: all of the roles below are initial proposals and must be accepted by the TOC per formation process_

- Proposed [Roles][sigroles] for SIG Observability
  - TOC Liaison: [Brendan Burns](https://github.com/brendandburns) (bburns@microsoft.com)
  - SIG Chairs: pending TOC guidance
    - _Nominated by the charter working group or self:_ Matt Young, Richard Hartmann, Steve Flanders, *Ricardo Aravena*\*
  - Tech Leads: pending TOC guidance

>     - _Nominated by the charter working group or self:_ Bartłomiej Płotka, *Richard Hartmann*\*, Michael Hausenblas, Alois Reitbauer

Reviewer: Nominate: Tigran Najaryan

<!-- TODO: please put github names here -->
[Matthew Young]:              https://github.com/halcyondude
[Alex Nauda]:                 @
[Alois Reitbauer]:            https://github.com/aloisreitbauer
[Bartłomiej (Bartek) Płotka]: https://github.com/bwplotka
[Daniel Khan]:                @
[Daniel Prata]:               @
[Lincoln Sward]:              @
[Matthias Loibl]:             https://github.com/metalmatze
[Michael Hausenblas]:         https://github.com/mhausenblas
[Ricardo Aravena]:            https://github.com/raravena80
[Richard Hartmann]:           https://github.com/RichiH
[Sergey Kanzhelev]:           @
[Steve Flanders]:             https://github.com/flands
[Ted Young]:                  @
[Tigran Najaryan]:            @
[Tommy Chong]:                https://github.com/techietommy
[Umair Ishaq]:                https://github.com/umairishaq

## Introduction

This document describes the purpose and operations of the Cloud Native
Computing Foundation ([CNCF]) Special Interest Group ([SIG]) on Observability.

This [SIG] focuses on topics pertaining to the observation
of [cloud native][cn-def] workloads. Additionally, it produces supporting

> material and best practices for end-users and provides guidance and

Reviewer 2: I'd still prefer "good practices" over "best practices". Software engineering (not even talking about cloud native yet) is in itself a very young discipline. While established disciplines such as civil engineering benefit from centuries of insights what (absolutely) worked and what not, they are indeed in a position to declare a certain practice as the "best". I do, with all optimism mustered, not believe we are in this position. We can, however, identify and describe good practices: "this worked for X in context Y". WDYT?

Reviewer 3: My 2 cents:
* I believe that it's super tiny detail (: 
* I am not a native English speaker, but I never heard the term "good" practice in this context. We always say "best practices". It does not mean "super super optimal and for sure the best, and there is no way there is better" :upside_down_face: 
* I believe we still aim for "best" practices even in the sense you described. I think about it this way: We are not the smartest in the world, but **to our knowledge** the X is the best practice in context of Y. If you think otherwise, tell us - we **would love to learn**. If the arguments win we move to Z, which will become new best practices! (: Can't see anything wrong with aiming for the best. 

My vote is to leave `best` as we are always aiming for best (best to our, CNCF and community knowledge).

coordination for CNCF projects working within the SIG’s scope.

> A full list of [CNCF projects][projs] can be found at [landscape.cncf.io].

Reviewer 3: Nice, it would be even nicer if the `https://landscape.cncf.io/ website would allow filtering through Categories where each SIG is a different category! :hugs:

Author: Perfect example of the kind of thing we can drive once the SIG is up and running :)

- Educate and help other CNCF projects in regarding observability techniques and
  practices available within the CNCF.
- Provide and maintain a vendor-neutral venue for relevant thought validation,
  discussion, and project feedback.
- Provide a ladder for community members to become involved with the technical
  oversight of projects within the SIG's scope in an open, transparent, and
  inclusive way.

## Areas considered in Scope

Observability focuses on patterns, projects, tools, and techniques related to
topics such as:

- Methodologies for instrumenting, collecting, processing, storing, querying,
  curating, and correlating observational data such as metrics, logging/events,
  trace spans, and profiling of cloud native workloads.
- Using distributed trace tooling to observe a series of calls between
  microservices to understand where time is being spent.
- Managing the complexity, operational cost, and resource consumption of
  observability tools and systems at the enterprise scale.
- Best practices for meaningful alerting, queries, and operational dashboards
  including how to manage things including rules, definitions, thresholds and
  policies.

> - How developers, operators, SRE, IT, and other actors comprehend, process, and

Reviewer 3: ```suggestion
- How developers, operators, site reliability engineers, IT engineers, and other actors comprehend, process, and
```
SRE might be not known to everyone (:

Author: good point, addressing in nits commit

  reason on distributed cloud-native systems.
- Projects that incorporate novel & insightful approaches to utilizing
  observability data such as:
  - ML, model training, Bayesian networks, and other data science techniques
    that enable anomaly & intrusion detection.

>   - correlating resource consumption with costing data to reduce the total cost

Reviewer 3: ```suggestion
  - Correlating resource consumption with costing data to reduce the total cost.
```

    of cloud native infrastructure
  - Using observability data exposed by service meshes, orchestrators, and other
    metric sources to inform continuous deployment tooling (e.g. Canary
    Predicates/Judges).
- Objective curation and generation of case studies pertaining to delivering
  observability tools/systems to end users.

> - Best practices around observability and its continuous improvement, e.g. post

Reviewer 3: I think this might be the same as ` Best practices for meaningful alerting, queries, and operational dashboards
  including how to manage things including rules, definitions, thresholds and
  policies.`

Maybe we can merge those to be more concise? (:

Author: If memory serves from discussion last week the distinction here is/was a focus on alerting/operational dashboards, vs. installing and configuring the systems themselves.  WDYT?  Agreed we could probably merge this to a common clause.

Reviewer 3: Cool, let's keep like it is for now then, iterate later (:

  mortems, runbooks
- Provide guidance around and foster interoperability between observability

>   solutions without trying to enforce one specific standard

Reviewer 3: ```suggestion
  solutions without trying to enforce one specific standard.
```

Author: +1

- Foster understanding of the prerequisites and corner-stones of observability
  like SLI/KPI, service objectives, and internal/external commitments.

The following is a non-exhaustive sample list of activities and deliverables

> that are in-scope for this SIG

Reviewer 3: ```suggestion
that are **in-scope* for this SIG:
```

Author: +1

- Summary and overview of projects available in the community.
- Catalog of reference architectures that draw from CNCF projects, combining
  them in useful and novel ways.
- Definitions of implementations and patterns for best practices for
  delivering observability tooling at enterprise scale.

> - Tooling composition and tool chain creation based on existing projects.

Reviewer 3: ```suggestion
- Tooling composition and toolchain creation based on existing projects.
```
:thinking:

Author: 👍 rolling it into a nits commit

- Best practices for operations and monitoring workflows using CNCF Projects.
- Organizing and helping to provide visibility to Meetups, Blogs, and Podcasts
  related to the scope of the SIG.
- Guidance for application development and architecture that is observable.
- Replicatable reference architectures.
- Patterns for observing application delivery pipelines.
- Education regarding instrumentation cloud native workloads.
- Processing and Accessing relevant observability data at scale.

> - Policy and security controls for observabilty data.

Reviewer 3: ```suggestion
- Policy and security controls for observability data.
```

Author: 👍 rolling it into a nits commit

- Creating artifacts as part of CI/CD pipelines that facilitate observation of

>   services.  Concrete examples might be:

Reviewer 3: ```suggestion
  services. Concrete examples might be:
```

Author: +1

>   - service profiles for Linkerd

Reviewer 3: ```suggestion
  - Service profiles for Linkerd.
```

Author: +1

>   - debug binaries or other diagnostic metadata.

Reviewer 3: ```suggestion
  - Debug binaries or other diagnostic metadata and tools.
```

Author: +1

>   - representative trace spans from failing CI tests.

Reviewer 3: ```suggestion
  - Representative trace spans from failing CI tests.
```

Reviewer 3: Also not sure about this one... It sounds like it related to debugging CI/CD not to debug service that is released or deployed :thinking:

## Areas considered out of Scope

Anything not explicitly considered in the scope above.

Examples include:

- Datastores that are not primarily used for observability. Those datastores
  might be in the scope of SIG Storage.
- Security aspects that need to be present when setting up cloud native
  infrastructure, these might be more relevant for SIG Security.
- How cloud native applications that need observability are deployed; this would

>   fall in the scope of SIG App Delivery

Reviewer 3: ```suggestion
  fall in the scope of SIG App Delivery.
```

Author: +1

- Contribute to [due diligence reports][ddr] to assist the CNCF TOC for projects
  in the scope of the SIG.
- Facilitate webinars and presentations from CNCF projects and domain experts in
  the scope of the SIG.
- Formation of [SIG working group(s)][sigwg] as resource capacity and member
  contribution allows.

  > _SIGs may choose to spawn focussed and time-limited working groups to achieve some of their responsibilities (for example, to produce a specific educational white paper, or portfolio gap analysis report). Working groups should have a clearly documented charter, timeline (typically a few quarters at most), and set of deliverables. Once the timeline has elapsed, or the deliverables delivered, the working group dissolves, or is explicitly re-chartered._

[ddr]: https://github.com/cncf/toc/blob/master/process/due-diligence-guidelines.md
[sigwg]: https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#responsibilities--empowerment-of-sigs

## Governance

- This SIG follows the [standard operating model][som] provided by the TOC
  unless otherwise stated here.

[som]: https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#operating-model

## Operations

> - Formation of the SIG follows the [documented process][sigform].

Reviewer 3: ```suggestion
- The formation of the SIG follows the [documented process][sigform].
```

Author: 👍 rolling it into a nits commit

> _Note: all of the roles below are initial proposals and must accepted by the TOC per formation process_

Reviewer 3: ```suggestion
_Note: all of the roles below are initial proposals and must be accepted by the TOC per formation process_
```

Author: 👍 rolling it into a nits commit

- Proposed [Roles][sigroles] for SIG Observability
  - TOC Liaison: *Jeff Brewer*\*
  - SIG Chairs: pending TOC guidance

>     - _Nominated by charter working group or self:_ Matt Young, Richard Hartmann, *Ricardo Aravena*\*

Reviewer 3: ```suggestion
    - _Nominated by the charter working group or self:_ Matt Young, Richard Hartmann, *Ricardo Aravena*\*
```

Author: 👍 rolling it into a nits commit

  - Tech Leads: pending TOC guidance

>     - _Nominated by charter working group or self:_ Bartłomiej Płotka, *Richard Hartmann*\*, Michael Hausenblas, Alois Reitbauer

Reviewer 3: ```suggestion
    - _Nominated by the charter working group or self:_ Bartłomiej Płotka, *Richard Hartmann*\*, Michael Hausenblas, Alois Reitbauer
```

Author: 👍 rolling it into a nits commit

# CNCF SIG Observability Charter

- [CNCF SIG Observability Charter](#cncf-sig-observability-charter)
  - [Introduction](#introduction)
  - [Mission](#mission)
  - [Areas considered in Scope](#areas-considered-in-scope)
  - [Areas considered out of Scope](#areas-considered-out-of-scope)
  - [Roadmap & Initial Efforts](#roadmap--initial-efforts)
  - [Governance](#governance)
  - [Operations](#operations)

*Initially authored by [Matthew Young][matthew young] with grateful review and
contributions from:
[Alex Nauda][Alex Nauda],
[Alois Reitbauer][Alois Reitbauer],

> [Bartłomiej (Bartek) Płotka][Bartłomiej (Bartek) Płotka],

Reviewer: ```suggestion
[Bartłomiej (Bartek) Płotka][Bartłomiej (Bartek) Płotka],
[Bogdan Drutu][Bogdan Drutu],
```

[Daniel Khan][Daniel Khan],
[Daniel Prata][Daniel Prata],
[Lincoln Sward][Lincoln Sward],
[Matthias Loibl][Matthias Loibl],
[Michael Hausenblas][Michael Hausenblas],
[Ricardo Aravena][Ricardo Aravena],
[Richard Hartmann][Richard Hartmann],
[Sergey Kanzhelev][Sergey Kanzhelev],
[Steve Flanders][Steve Flanders],
[Ted Young][Ted Young],
[Tigran Najaryan][Tigran Najaryan],
[Tommy Chong][Tommy Chong],

> and [Umair Ishaq][Umair Ishaq].*

Reviewer 4: ```suggestion
[Umair Ishaq][Umair Ishaq],
and [Yuri Shkuro][].*
```

<!-- TODO: please put github names here -->
[Matthew Young]:              https://github.com/halcyondude
[Alex Nauda]:                 @
[Alois Reitbauer]:            https://github.com/aloisreitbauer

> [Bartłomiej (Bartek) Płotka]: https://github.com/bwplotka

Reviewer: ```suggestion
[Bartłomiej (Bartek) Płotka]: https://github.com/bwplotka
[Bogdan Drutu]:                      http://github.com/bogdandrutu
```

[Daniel Khan]:                @
[Daniel Prata]:               @
[Lincoln Sward]:              @
[Matthias Loibl]:             https://github.com/metalmatze
[Michael Hausenblas]:         https://github.com/mhausenblas
[Ricardo Aravena]:            https://github.com/raravena80
[Richard Hartmann]:           https://github.com/RichiH
[Sergey Kanzhelev]:           @
[Steve Flanders]:             https://github.com/flands
[Ted Young]:                  @

> [Tigran Najaryan]:            @

Reviewer: ```suggestion
[Tigran Najaryan]:            https://github.com/tigrannajaryan
```

[Tommy Chong]:                https://github.com/techietommy

> [Umair Ishaq]:                https://github.com/umairishaq

Reviewer 4: ```suggestion
[Umair Ishaq]:                https://github.com/umairishaq
[Yuri Shkuro]:                https://github.com/yurishkuro
```

  inclusive way.

## Areas considered in Scope

Observability focuses on patterns, projects, tools, and techniques related to
topics such as:

- Methodologies for instrumenting, collecting, processing, storing, querying,
  curating, and correlating observational data such as metrics, logging/events,
  trace spans, and profiling of cloud native workloads.
- Using distributed trace tooling to observe a series of calls between
  microservices to understand where time is being spent.
- Managing the complexity, operational cost, and resource consumption of
  observability tools and systems at the enterprise scale.
- Best practices for meaningful alerting, queries, and operational dashboards
  including how to manage things including rules, definitions, thresholds and
  policies.
- How developers, operators, Site Reliability Engineers (SRE), IT Engineers, and
  other actors comprehend, process, and reason on distributed cloud-native
  systems.
- Projects that incorporate novel & insightful approaches to utilizing
  observability data such as:
  - ML, model training, Bayesian networks, and other data science techniques
    that enable anomaly & intrusion detection.

>   - Correlating resource consumption with costing data to reduce the total cost

Reviewer 5: 👍

_Note: all of the roles below are initial proposals and must be accepted by the TOC per formation process_

- Proposed [Roles][sigroles] for SIG Observability

>   - TOC Liaison: [Brendan Burns](https://github.com/brendandburns) (bburns@microsoft.com)

Reviewer 6: Woop woop!

Reviewer 3: :heart:
